src/main/java/ch/njol/skript/classes/data/DefaultComparators.java

```diff
@@ -670,6 +666,21 @@ public boolean supportsOrdering() {
 		Comparators.registerComparator(Color.class, Color.class, (one, two) -> Relation.get(one.asBukkitColor().equals(two.asBukkitColor())));
 		Comparators.registerComparator(Color.class, org.bukkit.Color.class, (one, two) -> Relation.get(one.asBukkitColor().equals(two)));
 		Comparators.registerComparator(org.bukkit.Color.class, org.bukkit.Color.class, (one, two) -> Relation.get(one.equals(two)));
+
+		if (Skript.classExists("org.bukkit.entity.EntitySnapshot")) {
+			boolean SNAPSHOT_AS_STRING_EXISTS = Skript.methodExists(EntitySnapshot.class, "getAsString");
+			Comparators.registerComparator(EntitySnapshot.class, EntitySnapshot.class, new Comparator<EntitySnapshot, EntitySnapshot>() {
+				@Override
+				public Relation compare(EntitySnapshot snap1, EntitySnapshot snap2) {
+					if (!snap1.getEntityType().equals(snap1.getEntityType()))
+						return Relation.NOT_EQUAL;
+					boolean isEqual = snap1.equals(snap2) || snap1.hashCode() == snap2.hashCode();
+					if (SNAPSHOT_AS_STRING_EXISTS)
+						isEqual = snap1.getAsString().equalsIgnoreCase(snap2.getAsString());
```

Comment:
```suggestion
					boolean isEqual = SNAPSHOT_AS_STRING_EXISTS
						? snap1.equals(snap2) || snap1.hashCode() == snap2.hashCode()
						: snap1.getAsString().equalsIgnoreCase(snap2.getAsString());
```

Reply:
why is hashCode even checked here?

Reply:
Just an fyi, you messed up this one. It should be `!SNAPSHOT_AS_STRING_EXISTS` or reorder the following conditions

Reply:
This shouldn't really be a ternary at all imo, just use an if else for legibility.
both are better options than the current code though.